docs/proposals/sysctl.md

```diff
@@ -0,0 +1,477 @@
+- `sysctls.alpha.kubernetes.io/pod-sysctl-2: kernel.msgmax:foo,bar`
+- ...
+
+The same annotation keys are used for pods and `PodSecurityPolicy` objects.
+
+The value is everything behind the colon.
+
+Note that the use of the enumerations for multiple keys is intentional. An alternative would be to encode a list of key/values pairs in one annotation. But as abritrary values are supposed to work, an encoded list is much harder to use. Another alternative would be encode the sysctl name, e.g. `netIpv6Idgen-delay` for `net.ipv6.idgen_delay` (neither dots nor underscore are valid characters in an annotation key). But that does not make it easier either.
+
+### Validation
+
+The name of each sysctl in `PodSecurityContext.Name` is validated against a static list of
+
+- specific sysctls
+- and a list of sysctl prefixes,
+
+all known to be namespaced by the kernel. These two lists are defined under `pkg/kubelet` and to be maintained by the nodes team.
+
+The initial whitelists will be:
+
+```go
+var whitelist = map[string]struct{}{
+       "kernel.msgmax":          {},
+       "kernel.msgmnb":          {},
+       "kernel.msgmni":          {},
+       "kernel.sem":             {},
+       "kernel.shmall":          {},
+       "kernel.shmmax":          {},
+       "kernel.shmmni":          {},
+       "kernel.shm_rmid_forced": {},
+}
+
+var whitelistPrefixes = []string{
+       "net.",
+       "fs.mqueue.",
+}
+```
+
+In addition to the whitelisting, the general format of the sysctl name will be checked:
+- 253 characters in length
+- it matches `sysctlRegexp`:
+
+```go
+const SysctlSegmentFmt string = "[a-z0-9]([_a-z0-9]*[a-z0-9])?"
+const SysctlFmt string = "(" + SysctlSegmentFmt + "\\.)*" + SysctlSegmentFmt
+var sysctlRegexp = regexp.MustCompile("^" + SysctlFmt + "$")
+```
+
+### Error behaviour
+
+Pods with specified sysctls are either launched with the given sysctl values or fail to launch, creating an event which is verbose enough to tell the user what happened.
+
+### Scheduling
+
+For the first step of sysctl support it is considered *out of scope* to schedule pods with certain sysctls onto certain hosts according to some given rules.
+
+One could imagine to offer certain non-namespaced sysctls as well which taint a host such that only containers with compatible sysctls settings are scheduled there. Moreover, there is no accounting of sysctl values if that makes any sense for a given sysctl.
+
+### SecurityContext Enforcement
+
+A list of permissible sysctls is to be added to `pkg/apis/extensions/types.go` (compare [security-context-constraints](security-context-constraints.md)):
+
+```go
+// PodSecurityPolicySpec defines the policy enforced.
+type PodSecurityPolicySpec struct {
+	...
+	// Sysctls is a white list of allowed sysctls in a pod spec. Each entry
+	// is either a plain sysctl name or ends in "*" in which case it is considered
+	// as a prefix of allowed sysctls.
+	Sysctls []string `json:"sysctls,omitempty"`
```

Comment:
As some sysctls might have severe consequences for the host when setting extreme values, it might make sense to not only allow sysctls in general, but only certain values or certain intervals.